src/ui/components/form/findComponent.tsx

```diff
@@ -34,10 +35,22 @@ export function findComponent(
       />
     );
   }
-  if (type.sub && !Array.isArray(type.sub)) {
-    const Component = findComponent(registry, type.sub, nestingNumber + 1);
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
-    return (props: any) => Vector({ Component, props: { ...props, nestingNumber, type } });
+  if (type.sub && type.type !== '[u8;4]') {
+    if (Array.isArray(type.sub)) {
+      const components = type.sub.map(
+        subtype =>
+          ({
+            Component: findComponent(registry, subtype, nestingNumber + 1),
+            name: subtype.name,
+          } as SubComponent)
+      );
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
+      return (props: any) => SubForm({ components, props: { ...props, nestingNumber, type } });
```

Comment:
Not clear why components and props are separated? Why not pass props as a spread and add on components array? This is what I did for enum

Reply:
I was thinking about higher order functions, you first pass the `components` to generate a `<SubForm>` component out of them by currying. Which then gets rendered. But I agree, its rather confusing, I already changed it!